kernelci/build.py

```diff
@@ -1228,6 +1246,21 @@ def run(self, jopt=None, verbose=False, opts=None):
         *verbose* is whether the build output should be shown
         """
         bmeta = self._meta.get('bmeta')
+        # Fetch linux firmware if this option enabled
+        # it means we need to have it ready to be embedded during build
+        if self._kernel_config_nonempty('CONFIG_EXTRA_FIRMWARE'):
+            print("Fetching firmware")
+            homedir = os.getenv("HOME")
```

Comment:
Using the `$HOME` directory is not a good approach, it should be done in the kernel tree or in the build output directory.

Reply:
Also it might be better to have a separate build step to copy the firmware files and update the config file, like `AddFirmware`.  So every time we run the command to build the kernel we don't go through the git clone step again (which btw will fail the 2nd time as there's already a repository, I believe there are functions to handle that already).

Reply:
Understood, i didn't thought that in some circumstances we might repeat build, i will add it as separate function and will add  handling if config option already exist and set and if firmware directory already exist, to not try to clone again.

Reply:
Suggestions implemented